eng/scripts/GenerateProjectList.ps1

```diff
@@ -1,10 +1,18 @@
 param(
     [switch]$ci
 )
-$ErrorActionPreference = 'stop'
 
+$ErrorActionPreference = 'stop'
+$msbuildEngine = 'dotnet'
 $repoRoot = Resolve-Path "$PSScriptRoot/../.."
 
-& "$repoRoot\eng\common\msbuild.ps1" -ci:$ci "$repoRoot/eng/CodeGen.proj" `
+try {
+  & "$repoRoot\eng\common\msbuild.ps1" -ci:$ci "$repoRoot/eng/CodeGen.proj" `
     /t:GenerateProjectList `
     /bl:artifacts/log/genprojlist.binlog
+} finally {
+  Remove-Item variable:global:_BuildTool -ErrorAction Ignore
+  Remove-Item variable:global:_DotNetInstallDir -ErrorAction Ignore
+  Remove-Item variable:global:_ToolsetBuildProj -ErrorAction Ignore
+  Remove-Item variable:global:_MSBuildExe -ErrorAction Ignore
```

Comment:
Copied this stuff from our /build.ps1 file because invoking tools.ps1 corrupts global state if this script, eng/scripts/GenerateReferenceAssemblies.ps1, or eng/scripts/CodeCheck.ps1 are invoked in an interactive PowerShell session.